src/main/kotlin/com/openlattice/data/DataGraphService.kt

```diff
@@ -490,13 +479,32 @@ open class DataGraphService(
                                 association.key.entityId to association.details
                             }.toMap()
                     val ids = doIntegrateEntities(
-                            entitySetId, entities, authorizedPropertiesByEntitySetId.getValue(entitySetId))
+                            entitySetId, entities, authorizedPropertiesByEntitySet.getValue(entitySetId))
                     entityKeyIds.putAll(ids)
                     entitySetId to ids.asSequence().map { it.key.entityId to it.value }.toMap()
                 }.toMap()
-        //Retrieve the src/dst keys
+
+        // Retrieve the src/dst keys (it adds all entitykeyids to mutable entityKeyIds)
         idService.getEntityKeyIds(entityKeys, entityKeyIds)
 
+        val edges = associations
+                .asSequence()
```

Comment:
It seems like this adds an extra iteration across `associations` vs having it above in the `.forEach`. I'm not sure what the scale of `associations` usually is so 🤷‍♂

Reply:
I'm not sure it's possible to avoid iterating over it twice and still ensure the src/dst types are allowed before writing any data -- but the scale is pretty large so if it is possible it would definitely be preferable